assembly/src/assembler/context.rs

```diff
@@ -21,6 +23,22 @@ pub struct AssemblyContext {
     allow_phantom_calls: bool,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum AssemblyContextError {
+    KernelError(KernelError),
+}
+
+impl Display for AssemblyContextError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            AssemblyContextError::KernelError(error) => write!(f, "{}", error),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for AssemblyContextError {}
```

Comment:
I'm not sure we need to introduce new error enum - we could probably just add a new variant to the `AssemblyError` (many methods here return `AssemblyError` already).

Reply:
replaced